bike_rent/models/sale.py

```diff
@@ -0,0 +1,36 @@
+from odoo import api, fields, models
+from datetime import timedelta
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_confirm(self):
+
+        def filter_rests(rec):
+            product = rec.product_id
+            return product.is_bike and product.type == 'service'
+
+        for record in self.order_line.filtered(filter_rests):
+            self.env['bike.rent'].create({
+                'notes': record.name,
+                'price': record.price_total,
+                'partner_id': record.order_partner_id.id,
+                'bike_id': record.product_id.id,
+                'rent_time': record.product_id.rent_duration,
+            })
+        return super(SaleOrder, self).action_confirm()
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    # Cannot be stored since depends on Datetime.now()
+    rent_end_date = fields.Datetime(string='Rent End Date', compute='_compute_rent_end_date', store=False)
+
+    @api.multi
+    def _compute_rent_end_date(self):
+        for line in self.filtered(lambda x: x.product_id.type == 'service' and x.product_id.is_bike):
```

Comment:
1. not too sure how should I proceed here, since the same filtering was used in a different model (`sale.order`, and this method is in `sale.order.line`). Do you mean I should've called that filtering method from the `sale.order` model? I guess I'm missing something really obvious here :|
2. Yeah, I totally forgot about generating reports post order confirmation (I was too focused on the idea that the main use of the report will be to provide it to the customer before signing/confirming the order). I don't believe related field with `bike.rent` would suite here, since the `bike.rent` record is only created after the order is confirmed (and we need to show the approximate date before that as well), so I split the method in two parts - if the order state is in _draft_ or _sent_ (both pre-confirmation states) we use the `Datetime.now()` option, and once the order is confirmed, we use the `rent_stop` value from the `bike.rent` record as suggested.